libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -561,6 +553,68 @@ inline _CUDA_VSTD::uint64_t * barrier_native_handle(barrier<thread_scope_block>
     return reinterpret_cast<_CUDA_VSTD::uint64_t *>(&b.__barrier);
 }
 
+
+_LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_DEVICE
```

Comment:
This should not be nodiscard, discarding the token is fine.

Reply:
I am going to keep nodiscard. Discarding the token should be done explicitly (preferably with a comment). The compiler diagnostic is extremely helpful in my opinion.